### PR TITLE
add contact/unarchive functionality

### PR DESCRIPTION
**Pull request by theandrewykim**
#### Why?
Looking to support the archive/unarchive functionality in our ruby wrapper:

https://developers.intercom.com/intercom-api-reference/reference#unarchive-a-contact

### CODE REVIEW

**Comment by theandrewykim** [comment]
Been having problems with bundler which looks to be the cause of the failures :(

File: spec/unit/intercom/contact_spec.rb [hunk 1]
```diff
@@ -260,6 +260,20 @@
...
 
+  describe 'archiving' do
+    let(:contact) { Intercom::Contact.new(id: '1', client: client) }
+
+    it 'should successfully archive a contact' do
+      client.expects(:post).with("/contacts/#{contact.id}/archive")
+      contact.archive
+    end
+
+    it 'should successfully unarchive a contact'
```
Comment (rahulgdsouza) [inline spec/unit/intercom/contact_spec.rb:271]:
```suggestion
    it 'should successfully unarchive a contact' do
```
This was causing bundler to fail

File: spec/unit/intercom/contact_spec.rb [hunk 2]
```diff
@@ -260,6 +260,20 @@
...
+      client.expects(:post).with("/contacts/#{contact.id}/archive")
+      contact.archive
```
Comment (rahulgdsouza) [inline spec/unit/intercom/contact_spec.rb:273]: Switch to `unarchive`
```
      client.expects(:post).with("/contacts/#{contact.id}/unarchive", {}).return(<Add expected response>)
      contact.unarchive
```

File: spec/unit/intercom/contact_spec.rb [hunk 3]
```diff
@@ -260,6 +260,20 @@
     end
   end
 
+  describe 'archiving' do
+    let(:contact) { Intercom::Contact.new(id: '1', client: client) }
+
+    it 'should successfully archive a contact' do
+      client.expects(:post).with("/contacts/#{contact.id}/archive")
```
Comment (rahulgdsouza) [inline spec/unit/intercom/contact_spec.rb:267]: This should be,
```
client.expects(:post).with("/contacts/#{contact.id}/archive", {}).return(<Add expected response>)
```

**Pull request closed without merging**